test/attr/attr_escaping.swift

```diff
@@ -217,3 +217,13 @@ class HasIVarCaptures {
     })()
   }
 }
+
+// https://bugs.swift.org/browse/SR-9760
+public enum FooEnum {
+  case bar
+}
+
+public protocol FooProtocol {
+  typealias CompletionHandler = (FooEnum) -> ()
```

Comment:
I think it makes sense to add a generic typealias here for completeness. Does it make sense to add a couple of methods in extension as well?

Also FYI, in general we are trying to reduce examples in a way where there are no unrelated things left e.g. access control and types which don't contribute to the problem, this example could further be reduced:

```swift
protocol SR_9760 {
  typealias F = () -> Void
  typealias G<T> = (T) -> Void
  func foo<T>(_: T, _: @escaping F) // ok
  func bar<T>(_: @escaping G<T>) // ok
}

extension SR_9760 {
  func fiz<T>(_: T, _: @escaping F) {} // ok
  func baz<T>(_: @escaping G<T>) {} // ok
}
```

Reply:
Done!